tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+
+**Application developers** - Responsible for designing, developing and maintaining application code. Developers use GitOps by checking files into their git repositories to configure GitOps settings to auto deploy and monitor their application and dependent infrastructure.
+
+**Site Reliability Engineers** - They leverage GitOps by checking in configuration files into their git repositories that add information for the customization of the workload, i.e. replica pod number, strategic merge solutions. 
+
+### Challenge(s) faced by the customer
+<!-- What challenges do the customer face? Why are they experiencing pain and why do current offerings not meet their need? -->
+Before the introduction of GitOps, enterprise application teams struggled with complex deployment processes; environment/application configuration drift; as well as difficulty following best practices for security and disaster recovery.  GitOps enables teams to define continuous deployment workflows and automation to simplify many of these tasks, which add significant value by increasing efficiency, security, creating better developer experiences, and allowing for faster deployments between code changes.
+
+Radius offers additional complementary value for enterprise application teams, including a rich, declarative application description which enables the Radius application graph.  Radius also provides a clear separate of concerns via Recipes.  However, without integration of Radius and GitOps, plus clear guidance on how to use the two together, enterprises already invested in GitOps don't know how to use both GitOps and Radius for a better together experience.
+
+<!-- What is the positive outcome for the customer if we deliver these features, i.e. what is the value proposition for the customer? Remember, this is customer-centric. -->
+
+## Key scenarios
+<!-- List ~3-7 high level scenarios to clarify the value of the Epic and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->
+
+### Scenario 1: As an operator I can install and configure Radius with my GitOps tool of choice
+<!-- One or two sentence summary -->
+My team uses GitOps to manage Kubernetes clusters and applications that run in those clusters.  All relevant configuration information is stored in a git repository that serves as the source of truth for both the application code and infrastructure configuration.  To use Radius, I need to include Radius as a dependency for my Kubernetes cluster, just as I would any other dependency, via my GitOps tools.
+
+### Scenario 2: As an operator or developer, I can deploy and manage Radius Environments, Recipes, applications, and resources via my GitOps toolset
+<!-- One or two sentence summary -->
+With Radius installed, as an Operator, I now define, deploy and manage a Radius environment and associate relevant Recipes to that environment. 
```

Comment:
I guess this is not just Recipes but also other Radius "objects" like Credentials etc.